feat(buildog): add status and slug column to blog table

furkanunsalan opened the pull request:
Edited the blog data table, added sorting, removed local storage usage and added dummyData Closes #61

kuzeykose commented on apps/buildog/src/app/(blog-pages)/blog/page.tsx:
```diff
@@ -1,85 +1,167 @@
-  [filename: string]: Blog;
-};
+// Define the Blogs type as an array of Blog objects
+type Blogs = Blog[];
```
You don't have to define this type. You can directly give Blog[] to useState type

kuzeykose commented on apps/buildog/src/app/(blog-pages)/blog/page.tsx:
```diff
@@ -1,85 +1,167 @@
-      if (typeof window !== "undefined") {
-        localStorage.setItem("blogs", JSON.stringify(updatedBlogs));
-      }
+      if (!prevBlogs) return prevBlogs;
```
Why we need this?

kuzeykose commented on apps/buildog/src/app/(blog-pages)/blog/page.tsx:
```diff
@@ -1,85 +1,167 @@
+      if (!prevBlogs) return prevBlogs;
+
+      // Filter out the blog to delete
+      const updatedBlogs = prevBlogs.filter((blog) => blog.title !== blogToDelete);
```
Filter may not a good idea here. We will make a backend endpoint for deleting. Right now no need a change.

kuzeykose replied:
Looking title to delete is not a best idea. We have to generate unique id to delete blog.

kuzeykose commented on .gitignore:
```diff
@@ -41,4 +41,4 @@ yarn-error.log*
 # main file generated by database script
-.main
\ No newline at end of file
+.main
```
is it .main? Should we use dot here?

I already add it by the way. You can remove this if you want.

furkanunsalan commented:
I updated the delete function to work with Blog IDs and fixed other requests, I left the delete logic (where I used filter method) as it is because it will be replaced once the backend for it is written.

kuzeykose reviewed (approved):
Awesome!
